### DISCUSSION TITLE
That "Firefox Privacy Notice" tab

### DISCUSSION THREAD

**Discussion by heiniovason** [mozilla/policy-templates#876] [Q&A] [answered]
Hi

I was wondering HOW does one make the "Firefox Privacy Notice" tab NOT show EVER ?
 
In my case I'm dealing with Ubuntu Desktop clients where the **/home/<user/** is wiped each time the session ends. Therefor the "Firefox Privacy Notice" tab will reappear on "initial" launch each time with a new user session. I would like to believe this can be handled globally through a policy, but I do not seem to be able to wrap my head around it. I had no luck with adding **UserMessaging** to the policies.json; maybe I did something wrong. Can't say for sure ..

This draft works really well for me, but I'm still battling the "Firefox Privacy Notice" tab 

```
#!/bin/bash

set -x

STARTPAGE="$1"
ADDITIONAL_PAGES="$2"
POLICIES_DIR="/usr/lib/firefox/distribution"
POLICY="policies.json"

if [ -z "$STARTPAGE" ]; then
  echo "WARNING: Missing <URL> argument. Not able to set Firefox startpage."
  exit 1
fi

PAGES_STRING=""
if [ -n "$ADDITIONAL_PAGES" ]; then
  IFS='|' read -ra PAGES_ARRAY <<< "$ADDITIONAL_PAGES"

  PAGES_STRING="\"Additional\": [" # start array-string
  for PAGE in "${PAGES_ARRAY[@]}"
  do
      PAGES_STRING+="\"$PAGE\","
  done
  PAGES_STRING=${PAGES_STRING::-1} # remove comma at end of list
  PAGES_STRING+="]," # finish array-string
fi

cat << EOF > "$POLICIES_DIR/$POLICY"
{
  "policies": {
    "Homepage": {
      "URL": "$STARTPAGE",
      "Locked": false,
      $PAGES_STRING
      "StartPage": "homepage"
    },
    "OverrideFirstRunPage": "",
    "OverridePostUpdatePage": ""
  }
}

EOF
```

**Comment by mkaply** [comment, marked as answer]
Sure.

If you set the datareporting.policy.dataSubmissionPolicyBypassNotification preference using the Preferences policy, you won't see that page.

https://github.com/mozilla/policy-templates/#preferences

**Reply by heiniovason** [reply]
Ma man. Much appreciated.

**UPDATED SCRIPT**

```
#!/bin/bash

set -x

STARTPAGE="$1"
ADDITIONAL_PAGES="$2"
POLICIES_DIR="/usr/lib/firefox/distribution"
POLICY="policies.json"

if [ -z "$STARTPAGE" ]; then
  echo "WARNING: Missing <URL> argument. Not able to set Firefox startpage."
  exit 1
fi

PAGES_STRING=""
if [ -n "$ADDITIONAL_PAGES" ]; then
  IFS='|' read -ra PAGES_ARRAY <<< "$ADDITIONAL_PAGES"

  PAGES_STRING="\"Additional\": [" # start array-string
  for PAGE in "${PAGES_ARRAY[@]}"
  do
      PAGES_STRING+="\"$PAGE\","
  done
  PAGES_STRING=${PAGES_STRING::-1} # remove comma at end of list
  PAGES_STRING+="]," # finish array-string
fi

cat << EOF > "$POLICIES_DIR/$POLICY"
{
  "policies": {
    "Homepage": {
      "URL": "$STARTPAGE",
      "Locked": false,
      $PAGES_STRING
      "StartPage": "homepage"
    },
    "OverrideFirstRunPage": "",
    "OverridePostUpdatePage": "",
    "Preferences": {
      "datareporting.policy.dataSubmissionPolicyBypassNotification": true
    }
  }
}

EOF
```

**Comment by mkaply** [comment]
If you end up using more prefs, you might need to switch to the newer pref format:

<code>
"Preferences": {
  "datareporting.policy.dataSubmissionPolicyBypassNotification": {
    "Value": true,
    "Status": "default"
  }
}
</code>

But what you have works fine.
